crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -2287,6 +2293,50 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn does_not_duplicate_global_registration_2311() {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, _http_proxy) = create_c8y_converter(&tmp_dir).await;
+
+        let mqtt_schema = MqttSchema::new();
+        let measurement_topic = mqtt_schema.topic_for(
+            &"device/main/service/my_measurement_service"
+                .parse()
+                .unwrap(),
+            &tedge_api::mqtt_topics::Channel::Measurement {
+                measurement_type: "my_type".to_string(),
+            },
+        );
+
+        let measurement_message = Message::new(&measurement_topic, r#"{"temperature": 21.37}"#);
+
+        let mut incoming_messages: VecDeque<Message> = vec![measurement_message].into();
+        let mut outgoing_messages = Vec::new();
+
+        while let Some(incoming_message) = incoming_messages.pop_front() {
+            let converted_messages = converter.convert(&incoming_message).await;
+            outgoing_messages.extend_from_slice(&converted_messages);
+
+            let converter_subs = &converter.config.topics;
+            converted_messages
+                .into_iter()
+                .filter(|m| converter_subs.accept(m))
+                .for_each(|m| incoming_messages.push_back(m))
+        }
```

Comment:
Why this while loop when the deque only contains a single message?

Reply:
In the last line with `for_each` I feed local registration messages back to `incoming_messages`, so both measurement message and auto-generated registration message are processed.